Confidence/src/main/java/com/spotify/confidence/Confidence.kt

```diff
@@ -57,6 +57,12 @@ class Confidence internal constructor(
         }
     }
 
+    fun setSdk(sdk: SdkMetadata) {
+        (flagResolver as RemoteFlagResolver).setSdk(sdk)
+        (eventSenderEngine as EventSenderEngineImpl).setSdk(sdk)
+        (flagApplierClient as FlagApplierClientImpl).setSdk(sdk)
+    }
```

Comment:
Is it really not possible to only support passing it in the `create()` function?

Having it mutable isn't really something we want to support, right?

Reply:
yes but then what would be the default? if someone wants to use confidence in conjunction with open feature, do we really want to force them to pass the correct Id or additional parameter when creating confidence? or we want to do it automatically?
another way is to pass the `apiKey` and other confidence parameters to the provider creation function and then provider to create the confidence.

Reply:
we can make it immutable but take into account that users of open feature, in order to have the correct id, should pass `SDK_ID_KOTLIN_PROVIDER`, and then my worry is, it will be easy for them to just not do it and our telemetry will be incorrect.

Reply:
Hmm... I didn't think about the OpenFeature provider case but purely for Flutter...

Can we explore different naming for create functions depending on the use case?

Or I wonder if we can pivot on the creation of the OF provider somehow to have it create the Confidence-instance behind the scenes or something? -- a bit strange since we likely want to also keep the reference to Confidence.

Reply:
> a bit strange since we likely want to also keep the reference to Confidence.
exactly!
for flutter, I already added in this PR the `sdk: SdkMetadata` parameter which can be passed in create functionn. `setSdk` is/will be called only within the provider context.